Example.md

```diff
@@ -0,0 +1,977 @@
+* When Pull Request is Pushed to master branch of repo `taxi`, it triggers [dev-cd-deploy-from-master-binding](#dev-cd-deploy-from-master-binding)/[dev-cd-deploy-from-master-template](#dev-cd-deploy-from-master-template)
+* When Pull Request is Opened/Synchronized on repo `taxi-stage-config`, it triggers [stage-ci-dryrun-from-pr-binding](#stage-ci-dryrun-from-pr-binding)/[stage-ci-dryrun-from-pr-template](#stage-ci-dryrun-from-pr-template)
+* When Pull Request is Pushed to master branch of repo `taxi-stage-config`, it triggers [stage-cd-deploy-from-push-binding](#stage-cd-deploy-from-push-binding)/[stage-cd-deploy-from-push-template](#stage-cd-deploy-from-push-template)
+
+For more information about GitHub REST API, please see https://developer.github.com/v3/activity/events/types/#pushevent
+
+Follow the these steps to create `cicd-event-listener`
+
+
+* Save the following content to `temp.yaml`
+
+```shell
+apiVersion: tekton.dev/v1alpha1
+kind: EventListener
+metadata:
+  name: cicd-event-listener
+spec:
+  serviceAccountName: demo-sa
+  triggers:
+    - name: dev-ci-build-from-pr
+      interceptors:
+        - cel:
+            filter: (header.match("X-GitHub-Event", "pull_request") && body.action == "opened" || body.action == "synchronize") && body.pull_request.head.repo.full_name == "$GITHUB_USER/taxi"
+      bindings:
+        - name: dev-ci-build-from-pr-binding
+      template:
+        name: dev-ci-build-from-pr-template
+    - name: dev-cd-deploy-from-master
+      interceptors:
+        - cel:
+            filter: (header.match("X-GitHub-Event", "push") && body.repository.full_name == "$GITHUB_USER/taxi") && body.ref.startsWith("refs/heads/master")
+      bindings:
+        - name: dev-cd-deploy-from-master-binding
+      template:
+        name: dev-cd-deploy-from-master-template
+    - name: stage-ci-dryrun-from-pr
+      interceptors:
+        - cel:
+            filter: (header.match("X-GitHub-Event", "pull_request") && body.action == "opened" || body.action == "synchronize") && body.pull_request.head.repo.full_name == "$GITHUB_USER/taxi-stage-config"
+      bindings:
+        - name: stage-ci-dryrun-from-pr-binding
+      template:
+        name: stage-ci-dryrun-from-pr-template
+    - name: stage-cd-deploy-from-push
+      interceptors:
+        - cel:
+            filter: (header.match("X-GitHub-Event", "push") && body.repository.full_name == "$GITHUB_USER/taxi-stage-config") && body.ref.startsWith("refs/heads/master")
+      bindings:
+        - name: stage-cd-deploy-from-push-binding
+      template:
+        name: stage-cd-deploy-from-push-template
+
+```
+
+* Run the following command to replace strings and apply yaml
+
+**_NOTE:_**  Make sure `GITHUB_USER` envrionment variable is set.
+
+```shell
+cat temp.yaml  | sed s"/\$GITHUB_USER/$GITHUB_USER/" | oc apply -f -
+```
```

Comment:
Something that isn't clear, is why sometimes use `${}` templating, and other times use `sed` ?

Reply:
I got rid of ${}